src/config.rs

```diff
@@ -334,6 +334,70 @@ impl Context {
     }
 }
 
+// Separate impl block for self address handling
+impl Context {
+    /// determine whether the specified addr maps to the/a self addr
+    pub async fn is_self_addr(&self, addr: &str) -> Result<bool> {
+        let addr = addr_normalize(addr).to_lowercase();
+
+        // The addresses we get here are already normalized and lowercase TODO is this true?
+        Ok(
+            self.get_config(Config::ConfiguredAddr).await?.as_deref() == Some(&addr)
+                || self.get_secondary_self_addrs().await?.contains(&addr),
+        )
+    }
+
+    /// Sets `primary_new` as the new primary self address and saves the old
+    /// primary address (if exists) as a secondary address.
+    pub(crate) async fn set_primary_self_addr(&self, primary_new: &str) -> Result<()> {
+        let primary_new = addr_normalize(primary_new).to_lowercase(); // TODO check if this might make problems
+
+        // Get all self addrs, including the old primary addr:
+        let mut secondary_new = self.get_all_self_addrs().await?;
+        // Remove the new primary addr in case it was a secondary addr before:
+        secondary_new.retain(|a| a != &primary_new);
+        self.set_config(
+            Config::SecondaryAddrs,
+            Some(secondary_new.join(" ").as_str()),
+        )
+        .await?;
+
+        self.set_config(Config::ConfiguredAddr, Some(&primary_new))
+            .await?;
+
+        Ok(())
+    }
+
+    /// Returns all primary and secondary self addresses.
+    pub async fn get_all_self_addrs(&self) -> Result<Vec<String>> {
+        let mut ret = Vec::new();
+
+        ret.extend(self.get_config(Config::ConfiguredAddr).await?.into_iter());
+        ret.extend(self.get_secondary_self_addrs().await?.into_iter());
+
+        Ok(ret)
+    }
+
+    /// Returns all secondary self addresses.
+    pub async fn get_secondary_self_addrs(&self) -> Result<Vec<String>> {
```

Comment:
wouldn't make this pub for now. `get_all_self_addrs()` should suffice.

Reply:
Thinking about it, we should just make both of them `pub(crate)` for now until there is a need (i.e. we add sth to the UI). No need to make anything public in case we might need it later (and then maybe noticing that that's not what we need).